src/python/WMCore/JobStateMachine/ChangeState.py

```diff
@@ -60,6 +60,28 @@ def discardConflictingDocument(couchDbInstance, data, result):
         return result
 
 
+def shrinkLargeFJR(couchDbInstance, sizeLimit=8e6):
+    """
+    Look at the CouchDB database queue and empty documents
+    with a 'fwjr' field larger than sizeLimit.
+    :param couchDbInstance: couchdb database instance
+    :param sizeLimit: integer with the limit number of bytes
+    :return: None
+    """
+    for doc in couchDbInstance._queue:
+        if sys.getsizeof(json.dumps(doc)) > sizeLimit:
+            if 'fwjr' in doc:
+                errMsg = f"The 'fwjr' attribute for jobid: {doc.get('jobid')} will "
+                errMsg += f"be empty because it is too large."
+                logging.warning(errMsg)
+                doc['fwjr'] = dict()
+            else:
+                errMsg = f"Found a large Couch document for job id: {doc.get('jobid')}, "
+                errMsg += "but it does not have a 'fwjr' field. Contact a developer!"
```

Comment:
what is a point of "Contact developer", i.e. what developer can do about it?

Reply:
This is a red flag for us, saying that something unexpected happen. A job report is meant to have a `fwjr` field!
Anyhow, I don't mind rephrasing it.